Sources/Build/ArtifactsArchiveMetadata.swift

```diff
@@ -0,0 +1,109 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Foundation
+import PackageModel
+import SPMBuildCore
+import TSCBasic
+import TSCUtility
+
+public struct ArtifactsArchiveMetadata: Equatable {
+    public let schemaVersion: String
+    public let artifacts: [String: Artifact]
+
+    public init(schemaVersion: String, artifacts: [String: Artifact]) {
+        self.schemaVersion = schemaVersion
+        self.artifacts = artifacts
+    }
+
+    public struct Artifact: Equatable {
+        let type: ArtifactType
+        let version: String
+        let variants: [Variant]
+
+        public init(type: ArtifactsArchiveMetadata.ArtifactType, version: String, variants: [Variant]) {
+            self.type = type
+            self.version = version
+            self.variants = variants
+        }
+    }
+
+    // In the future we are likely to extend the ArtifactsArchive file format to carry other types of artifacts beyond executables.
+    // Additional fields may be required to support these new artifact types e.g. headers path for libraries.
+    public enum ArtifactType: String, RawRepresentable,  Decodable {
+        case executable
+    }
+
+    public struct Variant: Equatable {
+        let path: String
+        let supportedTriplets: [Triple]
+
+        public init(path: String, supportedTriplets: [Triple]) {
+            self.path = path
+            self.supportedTriplets = supportedTriplets
+        }
+    }
+}
+
+extension ArtifactsArchiveMetadata {
+    public static func parse(fileSystem: FileSystem, rootPath: AbsolutePath) throws -> ArtifactsArchiveMetadata {
+        let path = rootPath.appending(component: "info.json")
+        guard fileSystem.exists(path) else {
+            throw StringError("ArtifactsArchive info.json not found at '\(rootPath)'")
+        }
+
+        do {
+            let bytes = try fileSystem.readFileContents(path)
+            return try bytes.withData { data in
+                let decoder = JSONDecoder.makeWithDefaults()
+                return try decoder.decode(ArtifactsArchiveMetadata.self, from: data)
+            }
+        } catch {
+            throw StringError("failed parsing ArtifactsArchive info.json at '\(path)': \(error)")
+        }
+    }
+}
+
+extension ArtifactsArchiveMetadata: Decodable {
+    enum CodingKeys: String, CodingKey {
+        case schemaVersion
+        case artifacts = "availableArtifacts"
```

Comment:
Nit: should this just be "artifacts"?  Are there any artifacts in the archive that aren't available?

Reply:
I just saw the example of the existing XCFrameworks format in the unit test, which uses AvailableLibraries, so now I understand where this comes from.  I could go either way on this: if we expect to move to `arar` as the future format that can handle all types of artifacts than maybe it's more important to pick names we think are clear than to be consistent with XCFrameworks.  Could go either way for me.